frontend/pages/browse.js

```diff
@@ -26,34 +30,52 @@ export default function Browse({
   filterChoices,
   hubs,
 }) {
+  // Initialize filters
   const [filters, setFilters] = useState({
     projects: {},
     members: {},
     organizations: {},
   });
   const [errorMessage, setErrorMessage] = useState("");
 
+  /**
+   * Fetches data from the server based on the newly provided
+   * filters. Returns an object with the new filter data, as well
+   * as other options.
+   *
+   * @param {string} type one of "projects", "members", "organizations"
+   * @param {Object} newFilters something like {"location": "", status: [], etc... }
+   * @param {boolean} closeFilters
+   * @param {string} oldUrlEnding previous end of URL through the query param
+   */
```

Comment:
Added some misc. comments throughout to help with debugging and what not. Happy to back / tweak any of them. 

This feature work made me really want to start incorporating TypeScript into the project after we build out more functionality and UX improvements.